deeplake/core/dataset/dataset.py

```diff
@@ -3073,6 +3073,7 @@ def _get_view(self, inherit_creds=True, creds: Optional[Dict] = None):
         )
 
         ds.index = Index()
+        ds.version_state = ds.version_state.copy()
```

Comment:
@farizrahman4u Is this line the fix to the bug that was causing the checkout problem?

Reply:
Yes.